### PR TITLE
Replace "redis" with "valkey" test code

### PR DESCRIPTION
**Pull request by Shivshankar-Reddy**
Occurrences of "redis" in TCL test suites and helpers, such as TCL client used in tests, are replaced with "valkey".

Occurrences of uppercase "Redis" are not changed in this PR.

No files are renamed in this PR.

### CODE REVIEW

**Comment by Shivshankar-Reddy** [comment]
@enjoy-binbin @zuiderkwast I have changed most of redis to valkey in this PR, I tried changing meaningfull names but it gave lot of trouble and test case failure so I went with replacing redis with "valkey". If you have some time could you please have a look at this?

File: src/cli_common.c
Comment (zuiderkwast) [inline src/cli_common.c]: This is not test code.

This is a breaking change. It is #198. See the discussions in #199.

We're not doing that. Besides, it is interfering with #199.

We already have had several PRs doing the same changes and we're reviewing them over and over.

Comment (Shivshankar-Reddy) [reply src/cli_common.c]: I got your point, But I had `benchmark: tls connecting using URI with authentication set,get` and` benchmark: connecting using URI with authentication set,get` test failed. that was the reason I changed, No problem we can wait until #199 (#198) gets merged.

Comment (zuiderkwast) [reply src/cli_common.c]: I'm not sure #198 will get merged at all.

Either way, I think we should test the redis:// protocol to make sure we support it. We want valkey-cli  8.0 to be compatible with valkey-server 7.2.x and with redis oss.

Comment (Shivshankar-Reddy) [reply src/cli_common.c]: Ohh.. Do we have any macro or check for backport compatibilty defined, I can go ahead update the commit in this pr or first I will commit them in different PR as they are going to be breaking changes and later we can focus on this PR changes. WDYT?

Comment (zuiderkwast) [reply src/cli_common.c]: I think just keep redis:// and rediss:// in this PR. Only update internal names in the test.

We are sure we want to support it and still test it.

We are not yet sure if we want to add another scheme. If we do, we want to test both.

Comment (Shivshankar-Reddy) [reply src/cli_common.c]: thank you for the details, sure I will change internally in test and commit again!

Comment (Shivshankar-Reddy) [reply src/cli_common.c]: Completed the changes and updated tests as well to pass now, when scheme is added we can change those test together. Thanks!

**Comment by enjoy-binbin** [comment]
full ci: https://github.com/valkey-io/valkey/actions/runs/8731463488

**Comment by Shivshankar-Reddy** [comment]
> full ci: https://github.com/valkey-io/valkey/actions/runs/8731463488

Thank you @enjoy-binbin , I will look into it and fix.

**Review by zuiderkwast — APPROVED** [review]
Very good. I will merge it before there are new merge conflicts. :)

Just to be clear what this PR is doing.

I can see only lowercase "redis" is replaced by "valkey".

It does not change "Redis" (in the output of `./runtest --help`) but we can do that in another PR.

It also does not rename any files. We can do that in another PR too. (Rename redis.tcl to valkey.tcl.)

**Comment by Shivshankar-Reddy** [comment]
> full ci: https://github.com/valkey-io/valkey/actions/runs/8731463488

Hi @enjoy-binbin , The failures happen on Mac OS workflows and I dont have Mac book so I have tried replicate the same tests on Ubuntu machine for multiple times and I could not replicate them.

From the logs we could see below tests failed.
1. We can read back the value we set before
2. Create a 5 nodes cluster

We may need help fom someone who has Mac and test this cases and see whether its related to CI or fails on manual run as well..

**Pull request merged**
